afs-storage-api/src/main/java/com/powsybl/afs/storage/InMemoryEventsBus.java

```diff
@@ -52,17 +53,19 @@ public void pushEvent(NodeEvent event, String topic) {
     public void flush() {
         lock.lock();
         try {
+            // to prevent the same thread to reenter the lock and modify the list
```

Comment:
I don't understand how the same thread could reenter the lock...
I think this method could be optimized since you make a copy. Why locking the whole function whereas you could just lock during the copy/clearing of the events list?